### ISSUE TITLE
Can't get stdout/stderr from a specific command (gcloud app deploy) on Windows.

### ISSUE THREAD

**Issue by thyb** [sindresorhus/execa#97]
Hello !

First, thanks a lot for your work on execa, it solves for me a big pain for some commands I wanted to spawn in my Electron application.

I have a strange issue trying to run `gcloud app deploy` with execa on Windows (it works perfectly on Mac OS X). I'm not sure if it's related to execa or directly Google Cloud Platform.
I can run all commands from [gcloud](https://cloud.google.com/sdk/gcloud/) with execa (`gcloud -v`, `gcloud projects list` etc...) but not `gcloud app deploy` (while it's working in cmd.exe). It sends me nothing - nothing on stdout - nothing on stderr - no promise resolved / rejected : Just nothing happens.

My current code looks something like this:

```
let bin = `gcloud${/^win/.test(process.platform) ? '.cmd' : ''}`
which(bin, (err, fullpath) => {
    let proc = execa(fullpath, ['app', 'deploy'], {
        cwd: appPath
    })
    proc.stdout.on('data', data => {
        parseDeploy(data.toString())
    })
    proc.stderr.on('data', data => {
        parseDeploy(data.toString())
    })
    proc.then(() => {
        ...
    }).catch(e => {
        ...
    })
})
```

I have tried lots of thing: 
- `execa()`
- `execa.shell()`
- options `shell:true`
- I tried `maxBuffer` to 1GB (just in case as it was the reason why I chose execa with the 10Mo default cache as it was crashing with `child_process.spawn()` on Mac OS X)
- It works with `detached:true` BUT I can't read stdout / stderr in realtime in the application

Does someone already got this issue ? I've searched around and found nothing promising.

**Comment by sindresorhus** [comment]
Does it work with `child_process.spawn()`?

**Comment by thyb** [comment]
I haven't found a way to work on Windows with child process...

Take a look at this Gist: https://gist.github.com/thyb/9b53b65c25cd964bbe962d8a9754e31f

This is the tests I have done on Windows. They work on Mac OS X.

Only the test `child_process-2-file.js` works completely on Mac OS X as it was my workaround for the 200kb buffer used in child_process - before discovering execa (gcloud executes during the deployment `npm install` which sends back a big chunk of string > 200kb - but I haven't reached this point on Windows so far).

**Comment by thyb** [comment]
Wow, I finally found an easy solution ! (I'm so happy)

Behind the scene, gcloud.cmd is running a python script. After reading tons of Node.js issue with ChildProcess / Python and Windows, I fell on this thread: https://github.com/nodejs/node-v0.x-archive/issues/8298

There is some known issue about running Python scripts from a Node.js Child Process.
They talk in this [comment](https://github.com/nodejs/node-v0.x-archive/issues/8298#issuecomment-54120669) about an unbuffered option for python. After updating the shell script in `gcloud.cmd` by adding the `-u` option, I noticed everything was working as expected

This [comment](https://github.com/nodejs/node-v0.x-archive/issues/8298#issuecomment-56892837) explains how to set this option as an environment variable (to not modify the windows shell script directly): https://docs.python.org/2/using/cmdline.html#envvar-PYTHONUNBUFFERED

So adding `PYTHONUNBUFERRED` to the environment variable fix this issue !

```js
execa(fullpath, ['app', 'deploy'], {
  cwd: appPath,
  env: Object.assign({}, process.env, {
    PYTHONUNBUFFERED: true
  })
})
```
